bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -84,6 +87,68 @@ determine_chart_version() {
         fail || return
 }
 
+assert_not_empty_list() {
+    run "$@"
+    assert_success || return
+    run jq_output length
+    assert_success || return
+    refute_output 0 || return
+}
+
+assert_true() {
+    run --separate-stderr "$@"
+    assert_success || return
+    is_true "$output" || return
+}
+
+# Given namespace and app name, assert that a log line contains the given string.
+assert_pod_log_line() {
+    local namespace="$1"
+    local selector="app=$2"
+    shift 2
+    local expect="$*"
+    run kubectl get pod --namespace "$namespace" --selector "$selector" --output=jsonpath='{.items[0].metadata.name}'
+    assert_success
+    assert_output || return
+    local name="$output"
+
+    run kubectl logs --namespace "$namespace" "$name"
+    assert_success || return
+    assert_output --partial "$expect" || return
+}
+
+# Pull down the image manually first so we are less likely to time out when
+# deploying rancher
+pull_rancher_image() {
+    local rancher_chart_version
+    if ! load_var rancher_chart_version; then
+        fail "Could not restore Rancher chart version"
+    fi
+    if using_docker; then
+        try docker pull --quiet "rancher/rancher:v$rancher_chart_version"
+    else
+        try nerdctl pull --namespace k8s.io --quiet "rancher/rancher:v$rancher_chart_version"
+    fi
```

Comment:
```suggestion
    local CONTAINERD_NAMESPACE=k8s.io
    try ctrctl pull --quiet "rancher/rancher:v$rancher_chart_version"
```

Also technically `--namespace` should come before the `pull` subcommand because it is a global option, but nowadays `nerdctl` deals with it correctly even when specified later.